handlers/ffmpeg/ffmpeg.go

```diff
@@ -0,0 +1,46 @@
+package ffmpeg
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/livepeer/catalyst-api/clients"
+	"github.com/livepeer/catalyst-api/config"
+	"github.com/livepeer/catalyst-api/errors"
+	"github.com/livepeer/catalyst-api/pipeline"
+)
+
+type HandlersCollection struct {
+	VODEngine *pipeline.Coordinator
+}
+
+// FFMPEG is called with something like the following:
+//
+//	ffmpeg -re -i SomeFile.mp4 -f hls -method PUT http://localhost:1234/<request id>/out.m3u8
+//
+// This HTTP handler is responsible for accepting that file and writing it out
+// to an external storage location
+func (h *HandlersCollection) NewFile() httprouter.Handle {
```

Comment:
How does it handle it if upload takes several mins but we get an onslaught of segments from ffmpeg in a short burst?

Reply:
From what I've seen with uploading ~700Mb files, it handles it fine and everything gets segmented and uploaded very quickly but I'll play around today with simulating a slow connection and understanding exactly how FFMPEG and this handler respond